WordPress/openverse-catalog
# Refactor Freesound to use ProviderDataIngester

## Fixes


Fixes #586 by @stacimc

## Description


This PR refactors the Freesound provider script to use the `ProviderDataIngester` class.

It's a pretty standard refactor, aside from some more unique functions that have additional retries built in.

One thing I noticed is that the original provider script is set up to allow processing a subset of data, but it has been run in such a way that it consumes the entire dataset. I think this is fine for now, but **as soon as we have a full, successful run**, this should be turned into a daily dated DAG. This will be a much more efficient way to run the DAG! Getting a successful run will require some additional error handling for failures that have occurred in the past, which I'm hoping to enumerate in another ticket.

## Testing Instructions

1. `just recreate && just test`
1. Run the DAG locally and verify that it respects whatever ingestion limit is set (note that there are 3 licenses it iterates through, so that ingestion limit will be hit 3 times)

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
